src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -296,5 +284,139 @@ internal SafeSslHandle(IntPtr validSslPointer, bool ownsHandle) : base(IntPtr.Ze
+                return bytesToCopy;
+            }
+
+            //Bio is already released by the ssl object
+            private void Dispose(bool isDisposing)
+            {
+                if (_handle.IsAllocated)
+                {
+                    _handle.Free();
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(false);
+                GC.SuppressFinalize(this);
+            }
+
+            ~ReadBioBuffer()
+            {
+                Dispose(false);
+            }
+        }
+
+        internal class WriteBioBuffer : IDisposable
+        {
+            private SafeBioHandle _bioHandle;
+            private GCHandle _handle;
+            private byte[] _byteArray;
+            private int _bytesWritten;
+            private bool _isHandshake;
+
+            internal WriteBioBuffer(SafeBioHandle bioHandle)
+            {
+                _bioHandle = bioHandle;
+                _handle = GCHandle.Alloc(this, GCHandleType.Normal);
+                Interop.CustomBio.BioSetGCHandle(_bioHandle, _handle);
+            }
+
+            public void SetBio(byte[] buffer, bool isHandshake)
+            {
+                _byteArray = buffer;
+                _bytesWritten = 0;
+                _isHandshake = isHandshake;
+            }
+
+            public int TakeBytes(out byte[] output)
+            {
+                var bytes = _bytesWritten;
+                output = _byteArray;
+                _bytesWritten = 0;
+                _byteArray = null;
+                return bytes;
+            }
+
+            public int Write(Span<byte> input)
+            {
+                if (_isHandshake)
+                {
+                    if (_byteArray == null)
+                    {
+                        _byteArray = new byte[input.Length];
+                        _bytesWritten = 0;
+                    }
+                    else if (_byteArray.Length - _bytesWritten < input.Length)
+                    {
+                        var oldSpan = new Span<byte>(_byteArray);
+                        _byteArray = new byte[input.Length + _bytesWritten];
+                        oldSpan.CopyTo(_byteArray);
+                    }
+                }
+                var bytesToWrite = Math.Min(input.Length, _byteArray.Length - _bytesWritten);
+                if (bytesToWrite < 1)
+                {
+                    Interop.Crypto.BioSetFlags(_bioHandle, Interop.Crypto.BIO_FLAGS.BIO_FLAGS_WRITE);
+                    return -1;
+                }
+                input.Slice(0, bytesToWrite).CopyTo(new Span<byte>(_byteArray, _bytesWritten));
+                _bytesWritten += bytesToWrite;
+                return bytesToWrite;
+            }
+
+            //Bio is already released by the ssl object
+            private void Dispose(bool isDisposing)
+            {
+                if (_handle.IsAllocated)
+                {
+                    _handle.Free();
```

Comment:
Yes [it does](https://github.com/dotnet/coreclr/blob/0350f97a49d1b4b51a874f61969cbeda2eff911a/src/mscorlib/src/System/Runtime/InteropServices/GcHandle.cs#L103); though since its a struct needs to be passed by ref or a class member to reflect it (e.g. single instance)